### DISCUSSION TITLE
Whether or not the blockscout is retrying data.

### DISCUSSION THREAD

**Discussion by kai-nakao** [blockscout/blockscout#7675] [Q&A]
I would like to know if blockscout retries when there is missing data. If you have details, I would be glad to hear the details.

**Comment by sl1depengwyn** [comment]
Could you please clarify your question, what specific data do you mean?

**Comment by kai-nakao** [comment]
The question is not about specific data, but if there are missing.

**Reply by sl1depengwyn** [reply]
I understand, what kind of data do you mean? Any? In this case it's too broad question. For instance, blocks are refetched by catchup indexer after realtime if realtime fails for any reason, token metadata is also refetching in case of errors

**Comment by kai-nakao** [comment]
Thank you, do you refetch internal transactions?

**Reply by sl1depengwyn** [reply]
Yes we do

**Reply by kai-nakao** [reply]
Thank you very much. I have a question when refetch missing internal transaction, do we explicitly go to get the internal transaction from rpc?

**Reply by vbaranov** [reply]
Internal transactions are fetched based on the `pending_block_operations` table. When the hash of the block appears in `pending_block_operations` table, sooner or later internal transactions for this block should be fetched.

**Reply by kai-nakao** [reply]
![pasted-2023 06 15-13 52 04](https://github.com/blockscout/blockscout/assets/65479384/e9552250-8efd-4876-9190-bd220bdee2da)
Thank you. However, the red underline record in the pending table is not disappearing and I cannot get the INTERNAL TRANSACTION, what is the cause and what is the possible solution?

**Reply by sl1depengwyn** [reply]
Do you have an archive node? Check the logs, there you may find errors related to internal transactions import for these blocks

**Reply by girafferz** [reply]
@sl1depengwyn 
thank you for your reply

Fetching internal transactions from archive nodes.

I have a question about the behavior of the pending_block_operations table.
If the block I add here already exists, won't it proceed?

in short

Manually remove the corresponding block from the blocks table
↓
in the pending_block_operations table
insert manually
↓
be recaptured
↓
internal transactions within that block are also fetched again

Will it move?

**Reply by sl1depengwyn** [reply]
Actually, all the blocks that are in pending_block_operations table already exist, so if you add some block that already exists blockscout would try to fetch internal transactions for this block

**Reply by girafferz** [reply]
@sl1depengwyn 

thank you for reply

I set
INDEXER_CATCHUP_MISSING_RANGES_BATCH_SIZE=2000000 (large enough)

I delete these tables  (block_number: 740121)
- blocks
- transactions
- internal_transactions

I manually create table 
- blocks (block_number: 740121,  refetch_needed: true, is_empty: true)

and after that
I manually create table
- pending_block_operations (block_number: 740121, )

and pending_block_operations record consumed (block_number: 740121)

but transactions and internal_transactions are blank


application log is below
```

2023-06-20T02:56:33.871 application=indexer fetcher=empty_blocks_to_refetch [info] Batch of empty blocks is sanitized
2023-06-20T02:56:36.033 application=indexer fetcher=block_catchup missing_block_count=0 shrunk=false [info] Index already caught up.
2023-06-20T02:56:36.034 application=indexer fetcher=block_catchup [info] Checking if index needs to catch up in 25000ms.
2023-06-20T02:56:43.872 application=indexer fetcher=empty_blocks_to_refetch [info] Start sanitizing of empty blocks. Batch size is 100
2023-06-20T02:56:43.883 application=indexer fetcher=empty_blocks_to_refetch [info] Batch of empty blocks is sanitized
2023-06-20T02:56:53.884 application=indexer fetcher=empty_blocks_to_refetch [info] Start sanitizing of empty blocks. Batch size is 100
2023-06-20T02:56:53.897 application=indexer fetcher=empty_blocks_to_refetch [info] Batch of empty blocks is sanitized
```

I want to regenerate transactions and internal transactions, but I can't, am I doing something wrong? I would appreciate it if you could teach me.

**Reply by sl1depengwyn** [reply]
If you delete block completely, you do not need to do anything, just wait and in some time catchup indexer will refetch this block, if you wait too long you can set `BLOCK_RANGES="740121..740121"`

**Reply by girafferz** [reply]
thank you so much
